src/ngraph/pass/core_fusion.cpp

```diff
@@ -73,3 +81,89 @@ void pass::CoreFusion::construct_relu()
     auto m = make_shared<pattern::Matcher>(max, callback);
     this->add_matcher(m);
 }
+
+void pass::CoreFusion::construct_folded_batch_norm()
+{
```

Comment:
this transformation seems to be geared towards MKLDNN since it's checking the rank of batchnorm. Does this really belong in CoreFusion at this point?

Reply:
Not really specific to MKLDNN but the replacement subgraph handles only 4-D cases, the way broadcasts are written.